src/CommandHandler.cxx

```diff
@@ -2,397 +2,407 @@
+  std::vector<StreamSettings> StreamSettingsList;
+  LOG(Sev::Info, "Command contains {} streams", StreamHDFInfoList.size());
+  for (auto &stream : StreamHDFInfoList) {
+    StreamSettings StreamSettings;
+    StreamSettings.StreamHDFInfoObj = stream;
+
+    json ConfigStream;
+    try {
+      ConfigStream = json::parse(stream.config_stream);
+    } catch (nlohmann::detail::parse_error const &e) {
+      LOG(Sev::Warning, "Invalid json: {}", stream.config_stream);
+      continue;
+    }
+
+    json ConfigStreamInner;
+    try {
+      ConfigStreamInner = ConfigStream.at("stream");
+    } catch (out_of_range const &e) {
       LOG(Sev::Notice, "Missing stream specification");
       continue;
     }
-    auto attributes = get_object(*stream.config_stream, "attributes");
-    auto &config_stream = *config_stream_value.v;
-    stream_settings.config_stream = config_stream_value.v;
-    LOG(Sev::Info, "Adding stream: {}", json_to_string(config_stream));
-    auto topic = get_string(&config_stream, "topic");
-    if (!topic) {
+
+    StreamSettings.ConfigStreamJson = ConfigStreamInner.dump();
+    LOG(Sev::Info, "Adding stream: {}", StreamSettings.ConfigStreamJson);
+
+    try {
+      StreamSettings.Topic = ConfigStreamInner.at("topic");
+    } catch (out_of_range const &e) {
       LOG(Sev::Notice, "Missing topic on stream specification");
       continue;
     }
-    stream_settings.topic = topic.v;
-    auto source = get_string(&config_stream, "source");
-    if (!source) {
+
+    try {
+      StreamSettings.Source = ConfigStreamInner.at("source");
+    } catch (out_of_range const &e) {
       LOG(Sev::Notice, "Missing source on stream specification");
       continue;
     }
-    stream_settings.source = source.v;
-    auto module = get_string(&config_stream, "writer_module");
-    if (!module) {
-      module = get_string(&config_stream, "module");
-      if (module) {
+
+    try {
+      try {
+        StreamSettings.Module = ConfigStreamInner.at("writer_module");
+      } catch (out_of_range const &e) {
+        // Allow the old key name as well:
+        StreamSettings.Module = ConfigStreamInner.at("module");
         LOG(Sev::Notice, "The key \"stream.module\" is deprecated, please use "
                          "\"stream.writer_module\" instead.");
-      } else {
-        LOG(Sev::Notice, "Missing key `writer_module` on stream specification");
-        continue;
       }
+    } catch (out_of_range const &e) {
+      LOG(Sev::Notice, "Missing key `writer_module` on stream specification");
+      continue;
     }
-    stream_settings.module = module.v;
-    bool run_parallel = false;
-    auto run_parallel_cfg = get_bool(&config_stream, "run_parallel");
-    if (run_parallel_cfg) {
-      run_parallel = run_parallel_cfg.v;
-      if (run_parallel) {
-        LOG(Sev::Info, "Run parallel {}", source.v);
-      }
+
+    try {
+      StreamSettings.RunParallel = ConfigStream.at("run_parallel");
+    } catch (out_of_range const &e) {
```

Comment:
As far as I know, empty catch is not bad per se, especially not here. Do you have something in mind which makes it a bad choice here?
The suggested alternative is harder to read and duplicates some code.

Reply:
I prefer the approach of only creating the exception in exceptional cases, rather than using it for logical control. In this context that means only calling `at` if you expect the key to exist. From a readability point of view it makes more sense to me to read "if the key exists then get the value" than "try to get the value but do nothing if it fails", but it is rather subjective. Do whichever you prefer, but if you keep the try-catch then I think the comment should say _why_ you are doing nothing in the `catch` rather than _that_ you are doing nothing, which is self evident.

Reply:
Exception handling is also logical control.
Performance is absolutely no concern, and it would require anyway profiling which one is better.
Also, the suggested `if` solution does not give a reason for the non-existent `else` branch either.
100% agree that this is subjective.

Reply:
Yes it is not about performance. "not give a reason for the non-existent else branch" - yes, but there isn't an empty `else` statement sitting there looking like a bug. You rightly put a comment there to show it is intentional that it contains no code, but as for any comment, it may as well be useful and say "why" not "what".